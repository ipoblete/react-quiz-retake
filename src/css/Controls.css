.form {
  text-align: center;
  padding: 20px;

  .title {
    font-weight: bold;
    padding: 10px;
  }

  .subtitle {
    font-weight: bold;
    padding: 10px;
  }
}