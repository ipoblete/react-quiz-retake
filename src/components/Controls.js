import React from 'react';
import PropTypes from 'prop-types';
import 'normalize.css';
import styles from '../css/Controls.css';

function Controls({ title, subtitle, onChange }) {
  return (
    <form className={styles.form}>
      <label className={styles.title}>Title:</label>
      <input type="text" name="title" value={title} onChange={onChange}/>
      <label className={styles.subtitle}>Subtitle:</label>
      <input type="text" name="subtitle" value={subtitle} onChange={onChange}/>
    </form>
  );
}

Controls.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Controls;
