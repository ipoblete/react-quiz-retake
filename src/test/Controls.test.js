import React from 'react';
import renderer from 'react-test-renderer';
import Controls from '../components/Controls';

describe('Controls', () => {
  it('matches a snapshot', () => {
    const controls = renderer.create(
      <Controls />
    ).toJSON();

    expect(controls).toMatchSnapshot();
  });
});
