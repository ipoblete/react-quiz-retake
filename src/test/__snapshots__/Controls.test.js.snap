// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls matches a snapshot 1`] = `
<form>
  <label>
    Title:
  </label>
  <input
    name="title"
    type="text"
  />
  <label>
    Subtitle:
  </label>
  <input
    name="subtitle"
    type="text"
  />
</form>
`;
